packages/algolia-search/hooks/useAlgoliaResults.ts

```diff
@@ -0,0 +1,53 @@
+import { useQuery } from '@graphcommerce/graphql'
+import { CurrencyEnum } from '@graphcommerce/graphql-mesh'
+import { ProductListItemFragment, ProductListItemProps } from '@graphcommerce/magento-product'
+import { StoreConfigDocument } from '@graphcommerce/magento-store'
+import { useHits } from 'react-instantsearch-hooks'
+import { AlgoliaHit } from '../lib/types'
+
+function hitsToProduct(items: AlgoliaHit[], currency?: string | null) {
+  const mapHits = items.map((item) => {
+    const currentCurrency = (currency ?? Object.keys(item.price)[0]) as CurrencyEnum
+    const price = item.price[currentCurrency]
+    const productUrlSplit = item.url.split('/')
+    const productUrl = productUrlSplit[productUrlSplit.length - 1]
+    const url_key = productUrl.substring(0, productUrl.length - 5)
```

Comment:
This should take the product_url_suffix in consideration. Can gotten from the storeconfig